common/src/main/java/io/netty/util/concurrent/PromiseTask.java

```diff
@@ -113,7 +147,8 @@ public final boolean trySuccess(V result) {
     }
 
     protected final boolean trySuccessInternal(V result) {
-        return super.trySuccess(result);
+        boolean done = super.trySuccess(result);
+        return clearTaskAfterCompletion(done, COMPLETED);
```

Comment:
nit: you can merge these lines

Reply:
sure, I actually only kept it as separate lines for readability